tools/serverpod_cli/lib/src/analyzer/models/validation/restrictions.dart

```diff
@@ -496,6 +499,27 @@ class Restrictions {
       ];
     }
 
+    if (def is ClassDefinition) {
+      var currentModel = parsedModels.findByClassName(def.className);
+
+      if (currentModel is ClassDefinition) {
+        var fieldWithDuplicatedName =
+            _findFieldWithDuplicatedName(currentModel, fieldName);
+        var parentClassWithDuplicatedFieldName =
+            _findAncestorWithDuplicatedFieldName(currentModel, fieldName);
+
+        if (fieldWithDuplicatedName != null &&
+            parentClassWithDuplicatedFieldName != null) {
+          return [
+            SourceSpanSeverityException(
+              'Duplicated field definition. Field with name "$fieldName" already exists in inherited class "${parentClassWithDuplicatedFieldName.className}".',
```

Comment:
Suggestion: Rename warning to: `The field name "$fieldName" is already defined in an inherited class ("${parentClassWithDuplicatedFieldName.className}").`.